Content.Client/Items/Systems/ItemSystem.cs

```diff
@@ -107,6 +109,10 @@ private bool TryGetDefaultVisuals(EntityUid uid, ItemComponent item, string defa
         layer.RsiPath = rsi.Path.ToString();
         layer.State = state;
         layer.MapKeys = new() { state };
+        if (TryComp(uid, out AtmosPipeColorComponent? pipeColor))
+        {
+            layer.Color = pipeColor.Color;
+        }
```

Comment:
This needs to be done elsewhere; it's not about the performance it's the fact that a common system is getting diluted by hard-coding in specific systems to it (which is why the eventbus exists, to avoid this).

Reply:
Now the color is obtained after the GetInhandVisualsEvent in PipeColorVisualizerSystem. The ItemSystem is now unchanged and does not depend on the Atmos Pipe Color Component.